### ISSUE TITLE
Update widget after change index value

### ISSUE THREAD

**Issue by valientegaston** [alitorki2651/flutter_gauge#3]
Hi! I'm using Provider as injector to change index value in realtime.  Index value changes, but not widget update happening.  Any clue? Thank you!

**Comment by madc** [comment]
I'm facing the same issue. The gauge animates to the initial issue and stays there regardless of any change on the index parameter.

**Comment by valientegaston** [comment]
I fixed it.  Just uncomment "eventObservable.add(widget.index);" inside build override method of _FlutterGaugeState class in flutter_gauge.dart file.

**Comment by carlosmcodes** [comment]
also facing the same issue. the suggestion above did not work for me :(

did anyone else figure out another route?

**Comment by svijay7598** [comment]
> I fixed it. Just uncomment "eventObservable.add(widget.index);" inside build override method of _FlutterGaugeState class in flutter_gauge.dart file.

yes this is worked for me ,I just uncomment the lines which are commented in flutter_gauge.dart file.

**Comment by adnandogar** [comment]
@svijay7598 uncomment all lines but gauge not updating its needle or its value.
is there any thing else should i do?

**Comment by virtuosonic** [comment]
> also facing the same issue. the suggestion above did not work for me :(
> 
> did anyone else figure out another route?

I uncommented all of _"flutter_gauge.dart"_ and then clicked _'Flutter Hot Restart'_ in Android Studio
